packages/world/test/KeysInTableModule.t.sol

```diff
@@ -0,0 +1,200 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import "forge-std/Test.sol";
+
+import { Schema, SchemaLib } from "@latticexyz/store/src/Schema.sol";
+import { SchemaType } from "@latticexyz/schema-type/src/solidity/SchemaType.sol";
+
+import { World } from "../src/World.sol";
+import { IBaseWorld } from "../src/interfaces/IBaseWorld.sol";
+import { ResourceSelector } from "../src/ResourceSelector.sol";
+import { ROOT_NAMESPACE } from "../src/constants.sol";
+
+import { CoreModule } from "../src/modules/core/CoreModule.sol";
+import { KeysInTableModule } from "../src/modules/keysintable/KeysInTableModule.sol";
+import { MODULE_NAMESPACE } from "../src/modules/keysintable/constants.sol";
+import { KeysInTable } from "../src/modules/keysintable/tables/KeysInTable.sol";
+import { getKeysInTable } from "../src/modules/keysintable/getKeysInTable.sol";
+import { getTargetTableSelector } from "../src/modules/utils/getTargetTableSelector.sol";
+
+contract KeysInTableModuleTest is Test {
+  using ResourceSelector for bytes32;
+  IBaseWorld world;
+  KeysInTableModule keysInTableModule = new KeysInTableModule(); // Modules can be deployed once and installed multiple times
+
+  bytes16 namespace = ROOT_NAMESPACE;
+  bytes16 sourceFile = bytes16("source");
+  bytes32 key1 = keccak256("test");
+  bytes32[] keyTuple1;
+  bytes32 key2 = keccak256("test2");
+  bytes32[] keyTuple2;
+
+  Schema sourceTableSchema;
+  Schema sourceTableKeySchema;
+  bytes32 sourceTableId;
+  bytes32 targetTableId;
+
+  function setUp() public {
+    sourceTableSchema = SchemaLib.encode(SchemaType.UINT256);
+    sourceTableKeySchema = SchemaLib.encode(SchemaType.BYTES32);
+    world = IBaseWorld(address(new World()));
+    world.installRootModule(new CoreModule(), new bytes(0));
+    keyTuple1 = new bytes32[](1);
+    keyTuple1[0] = key1;
+    keyTuple2 = new bytes32[](1);
+    keyTuple2[0] = key2;
```

Comment:
should we add a test for a key tuple with 2+ entries? (with the expectation of `hasTable` working for the full key, but `getKeysInTable` returning an array of the first key only)

Reply:
The flags an important point; the API of `getKeysInTable` is wrong. It only returns an array of `bytes32` (single keys instead of a composite array).